src/BloomBrowserUI/pageChooser/js/page-chooser.ts

```diff
@@ -0,0 +1,156 @@
+/// <reference path="../../bookEdit/js/getIframeChannel.ts" />
+
+window.addEventListener('message', process_EditFrame_Message, false);
+
+var logic;
+
+function process_EditFrame_Message(event: MessageEvent): void {
+
+    var params = event.data.split('\n');
+
+    switch(params[0]) {
+        case 'AddSelectedPage':
+            if(logic)
+                logic.addPageClickHandler();
+            else
+                alert("received AddSelectedPage message before PageChooser was created");
+            return;
+
+        case 'Data':
+            logic = new PageChooser(params[1]);
+            logic.LoadInstalledCollections();
+            return;
+
+        default:
+    }
+}
+
+// this version of the test string may be useful later testing more than one template collection
+//var JSONTestString = "[{ \"templateBookUrl\": \"../../../DistFiles/factoryCollections/Templates/Basic Book/Basic Book.htm\" }, { \"templateBookUrl\": \"../../../DistFiles/factoryCollections/Templates/Basic Book/Basic Book.htm\" }]";
+// no longer using test string, but let's keep it around as documentation of what PageChooser's ctor is expecting
+//var JSONTestString = "[{ \"templateBookUrl\": \"bloom/localhost/C$/BloomDesktop/DistFiles/factoryCollections/Templates/Basic Book/Basic Book.htm\" }]";
+
+class PageChooser {
+
+    private _templateBookUrls : string;
+    private _selectedTemplatePage : JQuery;
+
+    constructor(templateBookUrls: string) {
+        if(templateBookUrls) {
+            this._templateBookUrls = templateBookUrls;
+        } else {
+            console.log('Expected url in PageChooser ctor!');
+        }
+
+        this._selectedTemplatePage = undefined;
+    }
+
+    thumbnailClickHandler( div ) : void {
+        // 'div' is an .invisibleThumbCover
+        // Mark any previously selected thumbnail as no longer selected
+        if (this._selectedTemplatePage != undefined) {
+            $(this._selectedTemplatePage).removeClass('ui-selected');
+        }
+        // Select new thumbnail
+        this._selectedTemplatePage = $( div ).parent();
+        $(this._selectedTemplatePage).addClass('ui-selected');
+        // Display large preview
+        var caption = $( '#previewCaption');
+        caption.text($('.templatePageCaption', this._selectedTemplatePage).text());
+        caption.attr('style', 'display: block;');
+        $( '#preview' ).attr( 'src', $( this._selectedTemplatePage ).find('iframe').first().attr('src'));
+        // TODO: disable the main dialog one! $( '#addPageButton' ).prop( "disabled", false );
+    } // thumbnailClickHandler
+
+    addPageClickHandler() : void {
+        if (this._selectedTemplatePage == undefined || this._templateBookUrls == undefined) {
+            return null;
```

Comment:
Todo? Inform user? (especially if because nothing selected...though I think this should be rare, we will mostly have some default? @hatton Could we not come up with a default always and get rid of all the 'nothing selected' stuff?)